File: examples/CBuilder/Simple/ObjectSpace/DataSetAndReporting/dMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "dMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "BoldHandle"
#pragma link "BoldHandles"
#pragma link "BoldPersistenceHandle"
#pragma link "BoldPersistenceHandleDB"
#pragma link "BoldSubscription"
#pragma link "BoldSystemHandle"
#pragma link "BoldAbstractModel"
#pragma link "BoldActions"
#pragma link "BoldDBActions"
#pragma link "BoldHandleAction"
#pragma link "BoldModel"
#pragma link "BoldUMLModelLink"
#pragma link "BoldAbstractDatabaseAdapter"
#pragma link "BoldAbstractPersistenceHandleDB"
#pragma link "BoldDatabaseAdapterIB"
#pragma link "BoldIBDatabaseAction"
#pragma resource "*.dfm"
TdmMain *dmMain;
//---------------------------------------------------------------------------
__fastcall TdmMain::TdmMain(TComponent* Owner)
  : TDataModule(Owner)
{
}
//---------------------------------------------------------------------------

File: examples/Delphi/Simple/ObjectSpace/DataSetAndReporting/dMain.pas
unit dMain;

interface

uses
  Classes,
  Controls,
  Forms,
  ActnList,
  BoldSubscription,
  BoldHandle,
  BoldHandles,
  BoldSystemHandle,
  BoldPersistenceHandle,
  BoldPersistenceHandleDB,
  BoldModel,
  BoldUMLModelLink,
  BoldUMLRose98Link,
  BoldActions,
  BoldHandleAction,
  BoldDBActions, BoldAbstractModel, BoldIBDatabaseAction, DB, IBDatabase,
  BoldAbstractDatabaseAdapter, BoldDatabaseAdapterIB,
  BoldAbstractPersistenceHandleDB;

type
  TdmMain = class(TDataModule)
    bshMain: TBoldSystemHandle;
    stiMain: TBoldSystemTypeInfoHandle;
    bmoMain: TBoldModel;
    BoldUMLRoseLink1: TBoldUMLRoseLink;
    ActionList1: TActionList;
    BoldActivateSystemAction1: TBoldActivateSystemAction;
    BoldUpdateDBAction1: TBoldUpdateDBAction;
    BoldPersistenceHandleDB1: TBoldPersistenceHandleDB;
    BoldDatabaseAdapterIB1: TBoldDatabaseAdapterIB;
    IBDatabase1: TIBDatabase;
    BoldIBDatabaseAction1: TBoldIBDatabaseAction;
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  dmMain: TdmMain;

implementation

{$R *.DFM}

end.
